corehq/apps/custom_data_fields/static/custom_data_fields/js/custom_data_fields.js

```diff
@@ -287,6 +304,11 @@ hqDefine('custom_data_fields/js/custom_data_fields', [
                 .attr('value', $('#custom-fields-form .nav-tabs li:last').hasClass('active'))
                 .appendTo(customDataFieldsForm);
 
+            $('<input type="hidden">')
+                .attr('name', 'req_profile')
+                .attr('value', self.profile_required_for())
```

Comment:
hm i'm surprised this works as expected, don't we need to stringify this since `self.profile_required_for` is an observableArray?

Reply:
Yeah, it is surprising. I haven't fully worked out why but when the input is created and appended to the form the values are pulled out of the array. For example if the value of `profile_required_for` is `["web_user", "commcare_user"]` the input looks like this: `<input type=​"hidden" name=​"require_profile" value=​"web_user,commcare_user">​`